subsys/net/lib/download_client/src/download_client.c

```diff
@@ -65,8 +72,11 @@ static int socket_timeout_set(int fd, int type)
 		.tv_usec = (timeout_ms % 1000) * 1000,
 	};
 
+#if !defined (CONFIG_NET_SOCKETS_POSIX_NAMES)
+	LOG_INF("Configuring socket timeout (%d s)", (int32_t)timeo.tv_sec);
```

Comment:
What's the reason for this change, a different underlying type of `timeo.tv_sec` in POSIX/non-POSIX variant? IMO we could stick to a single approach with the cast, it should work in both cases.

Reply:
yes, at least with POSIX_API it is 64bit and original code resulted a warning: 
/home/jani/code_wa/ncs/nrf/subsys/net/lib/download_client/src/download_client.c:78:10: warning: format '%ld' expects argument of type 'long int', but argument 2 has type 'time_t' {aka 'long long int'} 

If remember right, casting was needed, %lld printing wasn't working. And a flagging it was a safe choice because didn't want to change the original.